### DISCUSSION TITLE
Is there a way to map returned values to their row/col position?

### DISCUSSION THREAD

**Discussion by thislooksfun** [eemeli/yaml#279] [Q&A] [answered]
I'm working on [yaml-enforce](https://github.com/thislooksfun/yaml-enforce), an api/cli to check structure and typing of YAML files, and I'd love to be able to give line+column numbers for validation errors. Is there a way to store that info somewhere while the data is being parsed?

Something like:
```js
import { parse } from "yaml";

const yamlText = "...";
const lineNumbers = {};
const data = parse(yamlText, {
  lineTracker: (path, line, column) => lineNumbers[path] = { line, col },
});
```

Where 'path' is preferably the dot-notation path to the value (like `foo.bar[2].baz`), though if I have to coerce it myself that's fine too.

---

Since LineCounter already exists I would also be fine with it just returning an offset into the input string as I can then use a LineCounter to calculate the line/col as needed.

```js
import { parse } from "yaml";

const yamlText = "...";
const offsets = {};
const data = parse(yamlText, {
  offsetTracker: (path, offset) => offsets[path] = offset,
});
```

**Comment by eemeli** [comment, marked as answer]
The intended way to achieve this would be something like this:

```js
import { LineCounter, parseDocument, visit } from 'yaml'

const lc = new LineCounter()
const doc = parseDocument('...', { lineCounter: lc })
visit(doc, (_key, node, path) => console.log({ path, pos: lc.linePos(node.range[0]) }))
const data = doc.toJS()
```
